app/src/main/java/com/example/healthinspector/CachedLists.java

```diff
@@ -5,34 +5,46 @@
 import android.content.Context;
 import android.util.Log;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.HashMap;
 
 public class CachedLists{
 
-    public static JSONObject loadAdditives(Context context) throws JSONException {
-        String json = "";
-        try {
-            InputStream is = context.getAssets().open("additives.json");
-            int size = is.available();
-            byte[] buffer = new byte[size];
-            is.read(buffer);
-            is.close();
-            json = new String(buffer, "UTF-8");
-        } catch (IOException ex) {
-            ex.printStackTrace();
-            return null;
+    private static CachedLists cachedLists = null;
+    private HashMap<String, String> additives = null;
+    private HashMap<String, String> allergens = null;
+
+    private CachedLists(Context context) throws JSONException, JsonProcessingException {
+        additives = loadJSONObject(context, "additives.json");
+        allergens = loadJSONObject(context, "allergens.json");
```

Comment:
It'd be better to load these in the getAdditives/getAllergens function if the respective object is null. This way we wont load the data until it is actually needed

Reply:
Also we should use named constants here